app/src/main/java/com/sunasterisk/boringweather/ui/search/SearchFragment.kt

```diff
@@ -0,0 +1,75 @@
+package com.sunasterisk.boringweather.ui.search
+
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.setFragmentResult
+import com.sunasterisk.boringweather.R
+import com.sunasterisk.boringweather.base.BaseFragment
+import com.sunasterisk.boringweather.data.model.City
+import com.sunasterisk.boringweather.di.Injector
+import com.sunasterisk.boringweather.ui.main.findNavigator
+import com.sunasterisk.boringweather.util.showToast
+import kotlinx.android.synthetic.main.fragment_search.*
+
+class SearchFragment : BaseFragment(), SearchContract.View {
+    override val layoutResource: Int = R.layout.fragment_search
+
+    override var presenter: SearchContract.Presenter? = null
+    override fun showSearchResult(cities: List<City>) {
+        (recyclerSearchCity.adapter as? CityAdapter)?.submitList(cities)
+    }
+
+    override fun showError(errorStringRes: Int) {
+        requireContext().showToast(getString(errorStringRes))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        presenter = SearchPresenter(this, Injector.getCityRepository(requireContext()))
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initView()
+        requestEditTextSearchFocus()
+    }
+
+    private fun initView() {
+        textField.setStartIconOnClickListener {
+            findNavigator()?.popBackStack()
+        }
+
+        textField.setEndIconOnClickListener {
+            editTextSearch.text?.toString()?.let { presenter?.searchCity(it) }
+        }
+
+        editTextSearch.setOnEditorActionListener { v, actionId, event ->
+            when (actionId) {
+                EditorInfo.IME_ACTION_SEARCH -> {
+                    v.text?.toString()!!.let { presenter?.searchCity(it) }
```

Comment:
```suggestion
                    v.text?.toString()?.let { presenter?.searchCity(it) }
```

Reply:
Nếu `v.text.toString()` thì có trường hợp nó tìm city "null"

Reply:
Thế này có đúng logic không nhỉ
```suggestion
                    v.text?.let { presenter?.searchCity(it.toString()) }
```